containers/src/Kokkos_DualView.hpp

```diff
@@ -410,18 +410,7 @@ class DualView : public ViewTraits<DataType, Properties...> {
   ///   typename dual_view_type::t_host hostView = DV.view<host_device_type> ();
   /// \endcode
   template <class Device>
-  KOKKOS_INLINE_FUNCTION const typename std::conditional_t<
-      impl_device_matches_tdev_device<Device>::value, t_dev,
-      typename std::conditional_t<
-          impl_device_matches_thost_device<Device>::value, t_host,
-          typename std::conditional_t<
-              impl_device_matches_thost_exec<Device>::value, t_host,
-              typename std::conditional_t<
-                  impl_device_matches_tdev_exec<Device>::value, t_dev,
-                  typename std::conditional_t<
-                      impl_device_matches_tdev_memory_space<Device>::value,
-                      t_dev, t_host>>>>>
-  view() const {
+  KOKKOS_INLINE_FUNCTION auto view() const {
```

Comment:
While you're at it, `inline` is implicit already:
```suggestion
  KOKKOS_FUNCTION auto view() const {
```